builder/builder-next/controller.go

```diff
@@ -131,6 +133,7 @@ func newSnapshotterController(ctx context.Context, rt http.RoundTripper, opt Opt
 			"inline":   inlineremotecache.ResolveCacheExporterFunc(),
 			"local":    localremotecache.ResolveCacheExporterFunc(opt.SessionManager),
 			"gha":      gha.ResolveCacheExporterFunc(),
+			"s3":       s3remotecache.ResolveCacheExporterFunc(),
```

Comment:
Comparing these two, I noticed that these are not "symmetrical";

- ❓ is it expected that the `inline` cache is included in `Exporters`, but not in `Importers` ?
- (nit): perhaps we should sort both lists have the exporters/importers in the same order (we could just sort them alphabetically, then at least there's no bike-shedding about what order they should appear in). 


<img width="743" alt="Screenshot 2023-03-06 at 12 24 31" src="https://user-images.githubusercontent.com/1804568/223098129-7856f4e0-aefa-4aab-b2de-ee21109a5fa5.png">

Reply:
Yes, I was wondering if it was something like that 🤔 it was confusing me a bit though (we could even consider including a comment to describe that, if what you linked to is indeed the expected behaviour). /cc @crazy-max @jedevc (if you know)

Reply:
Yeah this is expected - they are assymetrical. Inline cache works fundamentally differently than the other types of cache though they expose the same interface (but we can import them in the same way).

Reply:
I would really prefer to keep s3/azure caching out of the main dockerd binary.